### ISSUE TITLE
Perspective Broker's NewStyleTests should use IOPump

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8793]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)| @markrwilliams reported|
|-|-|
|Trac ID|trac#8793|
|Type|enhancement|
|Created|2016-08-27 23:36:09Z|

`twisted.spread.test.test_pb.NewStyleTests` uses an actual TCP transport for all its tests.  Instead it should use an `IOPump` like, say, `BrokerTests`.

This would make it easier to address non-determinism in garbage collection, enabling the tests to pass on PyPy.

<details><summary>Searchable metadata</summary>

```
trac-id__8793 8793
type__enhancement enhancement
reporter__markrwilliams markrwilliams
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1472340969754625 1472340969754625
changetime__1472370309036739 1472370309036739
version__None None
owner__markrwilliams markrwilliams

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_1"></a>|@markrwilliams commented|
|-|-|

PyPy isn't actually affected by this particular test.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)|@rodrigc set owner to @markrwilliams|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc commented|
|-|-|

https://github.com/twisted/twisted/pull/530#issuecomment-242959675

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_5"></a>|@markrwilliams set status to `closed`|
|-|-|

In changeset fb29a1cc14c15203e0f64ecc99e5225ed753e615
```
#!CommitTicketReference repository="" revision="fb29a1cc14c15203e0f64ecc99e5225ed753e615"
Merge pull request #530 from twisted/8793-pb-newstyletests-iopump

Merge 8793-pb-newstyletests-iopump: Make Perspective Broker's NewStyleTests use IOPump

Author: markrwilliams
Reviewer: rodrigc
Fixes: #8793

Introduce twisted.spread.test.test_pb.connectServerAndClient, which takes a PBClientFactory and PBServerFactory and connects the objects obtained from the two via an IOPump.  NewStyleTests now uses this instead of a TCP connection over the loopback interface.
```
